src/config/diff.rs

```diff
@@ -15,6 +20,8 @@ impl ConfigDiff {
 
     pub fn new(old: &Config, new: &Config) -> Self {
         ConfigDiff {
+            #[cfg(feature = "api")]
+            api: api::Diff::from_api(&old.api, &new.api),
```

Comment:
nit: All the other ones are called `Difference`, so maybe use a similar name for `api::Diff`?

Reply:
I'm also unsure if we _want_ to expose the api diff through the `ConfigDiff` type?

I don't know enough about this part of the code-base to reason one way or the other, but it does seem like right now it only exposes diffs for sources/transforms/sinks, while there are other top-level Vector configuration options as well that could differ, but aren't included in this type?

Reply:
👍 Will rename to `Difference`.

I'm piggybacking `ConfigDiff` because it seemed like a catch-all for any configuration changes that affect topology.

I could introduce a separate type here, although I'm not clear if that would add much benefit; I like the fact we can detect config differences in one place rather than splitting it up.

What might be better though is not having to do this 'raw' match on `api::Diff` in topology, and instead have convenience methods on the diff itself to detect whether it's a Start|Restart|Stop. That would avoid needing to bring in separate types to discern changes.

Reply:
Updated in 11b98056edd574979318b912f4a96bcaa9019f45.

Added some convenience methods to save some matching on the 'inner' `api::Difference` enum, in places:

```rust
#[cfg(feature = "api")]
let api = match &diff.api {
    Some(api_diff) if api_diff.is_start_or_restart() => {
        Some(Server::new(config.api.bind.unwrap()))
    }
    _ => None,
};
```